projects/distributed-tracing/src/shared/services/filter-builder/graphql-filter-builder.service.ts

```diff
@@ -14,7 +14,7 @@ export class GraphQlFilterBuilderService {
         new GraphQlFieldFilter(
           filter.field,
           this.toGraphQlOperator(filter.operator),
-          this.toGraphQlArgumentValue(filter.metadata.type, filter.value)
+          this.toGraphQlArgumentValue(filter.metadata.type as AttributeMetadataType, filter.value)
```

Comment:
this is that cast back - if we correctly type FilterAttribute, then no cast should be required